make format validate by default (#1553)
json-schema-org/json-schema-spec: specs/jsonschema-validation.md

correctly utilize certain values. The `format` annotation keyword is defined to
allow schema authors to convey semantic information for a fixed subset of values
which are accurately described by authoritative resources, be they RFCs or other
external specifications. Format values defined externally to this document
SHOULD also be based on such authoritative resources in order to foster
interoperability.

The value of this keyword MUST be a string. While this keyword can validate any
type, each distinct value will generally only validate a given set of instance
types. If the type of the instance to validate is not in this set, validation
for this keyword SHOULD succeed. All format values defined in this section apply
to strings, but a format value can be specified to apply to any instance types

> defined in the data model defined in the [core JSON Schema](#json-schema)[^1].

Reviewer: This sentence sounds incomplete. How about something like ...

```suggestion
defined in the data model defined in the [core JSON Schema](#json-schema) specification[^1].
```

(I realize this isn't something you changed, but it's something we can clean up while we're here.)

[^1]: Note that the `type` keyword in this specification defines an "integer"
type which is not part of the data model. Therefore a format attribute can be
limited to numbers, but not specifically to integers. However, a numeric format
can be used alongside the `type` keyword with a value of "integer", or it could
be explicitly defined to always pass if the number is not an integer, which
produces essentially the same behavior as only applying to integers.

> Implementations SHOULD provide assertion behavior for the format values defined
> by this document[^2] and MUST refuse to process any schema which contains an
> unsupported format value.

Reviewer: I feel like it's people might think it's unclear whether "unsupported" mean unsupported by the spec or unsupported by the implementation. I think a small change can make that less ambiguous.

```suggestion
Implementations SHOULD provide assertion behavior for the format values defined
by this document[^2] and MUST refuse to process any schema which contains an format value it doesn't support.
```

Author: If you can edit the wrapping for that, I'll just commit it from here.

from previous iterations of this specification. Previously, `format` was an
annotation-only keyword by default and implementations that supported assertion
were required to offer some configuration that allowed users to explicitly
enable assertion. Assertion is now a requirement in order to meet user
expectations. See [json-schema-org/json-schema-spec #1520](https://github.com/json-schema-org/json-schema-spec/issues/1520) for more.

In addition to the assertion behavior, this keyword also produces its value as
an annotation.

Implementations:

- SHOULD provide validation for each format attribute defined in this
  document;
- MAY support format values not defined in this document, but such support MUST
  be configurable and disabled by default;
- SHOULD use a common parsing library or a well-known regular expression for
  each format;
- SHOULD clearly document how and to what degree each format attribute is
  validated.

The requirement for validation of format values in general is limited to
syntactic checking; implementations SHOULD NOT attempt to send an email, connect
to a URL, or otherwise check the existence of an entity identified by a format
instance.

> #### Custom format attributes
> 
> Implementations MAY support custom format attributes. Save for agreement between
> parties, schema authors SHALL NOT expect a peer implementation to support such
> custom format attributes.

Reviewer: We removed the part where it defines the term "format attributes". We shouldn't use it here.

```suggestion
#### Custom format values

Implementations MAY support custom format values. Save for agreement between
parties, schema authors SHALL NOT expect a peer implementation to support such
custom format values.
```

Author: Good catch.
